modules/RsIskraMeter/src/main.rs

```diff
@@ -0,0 +1,1034 @@
+    ) -> Result<f64> {
+        let exp_register = self.read_input_registers_fixed::<1>(exp_register_address)?;
+        // Signed Value (16 bits)
+        let var = self.read_input_registers_fixed::<2>(counter_register_address)?;
+        Ok(counter(var, exp_register[0]))
+    }
+
+    fn read_meter_value(&self) -> Result<Powermeter> {
+        let resp = Powermeter {
+            var: Some(ReactivePower {
+                l_1: Some(self.read_t6(150)?),
+                l_2: Some(self.read_t6(152)?),
+                l_3: Some(self.read_t6(154)?),
+                total: self.read_t6(148)?,
+            }),
+            current_a: Some(Current {
+                dc: Option::None,
+                l_1: Some(self.read_t6(126)?),
+                l_2: Some(self.read_t6(128)?),
+                l_3: Some(self.read_t6(130)?),
+                n: Option::None,
+            }),
+            energy_wh_export: Some(Energy {
+                l_1: Option::None,
+                l_2: Option::None,
+                l_3: Option::None,
+                total: self.read_counter(415, 420)?,
+            }),
+            energy_wh_import: Energy {
+                l_1: Option::None,
+                l_2: Option::None,
+                l_3: Option::None,
+                total: self.read_counter(414, 418)?,
+            },
+            // TODO(kch) why freq values are different?
+            frequency_hz: Some(Frequency {
+                l_1: self.read_t5(105)?,
+                l_2: Option::None,
+                l_3: Option::None,
+            }),
+            // TODO(kch) meter_id
+            meter_id: Option::None,
+            phase_seq_error: Option::None,
+            power_w: Some(Power {
+                l_1: Some(self.read_t6(142)?),
+                l_2: Some(self.read_t6(144)?),
+                l_3: Some(self.read_t6(146)?),
+                total: self.read_t6(140)?,
+            }),
+            timestamp: Utc::now().to_rfc3339(),
+            voltage_v: Some(Voltage {
+                dc: Option::None,
+                l_1: Some(self.read_t5(107)?),
+                l_2: Some(self.read_t5(109)?),
+                l_3: Some(self.read_t5(111)?),
+            }),
+            current_a_signed: None,
+            energy_wh_export_signed: None,
+            energy_wh_import_signed: None,
+            frequency_hz_signed: None,
+            power_w_signed: None,
+            signed_meter_value: None,
+            var_signed: None,
+            voltage_v_signed: None,
+        };
+        Ok(resp)
+    }
+
+    fn write_metadata(&self, evse_id: &str) -> Result<()> {
+        // Warning: JSON names must be in specified order and without whitespaces.
+        let message: String = format!("{{\"FV\":\"1.0\",\"GI\":\"\",\"GS\":\"\",\"PG\":\"\",\"MV\":\"\",\"MM\":\"\",\"MS\":\"\",\"MF\":\"\",\"IS\":true,\"IF\":[\"RFID_PLAIN\"],\"IT\":\"NONE\",\"ID\":\"{} {}\",\"CT\":\"EVSEID\",\"CI\":\"vestel\",\"RD\":[]}}", evse_id, create_random_meter_session_id());
```

Comment:
This string contains a lot hard coded values. Especially "vestel" doesn't make sense here.